examples/ASMapNode/Sample/MapHandlerNode.m

```diff
@@ -109,90 +118,90 @@ - (void)didLoad
   [self addAnnotations];
 }
 
-#pragma mark - Layout
-
-- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
-{
+/** 
+ * ---------------------------------ASStackLayoutSpec--------------------------------
+ * |  ------------------------------ASInsetLayoutSpec-----------------------------  |
+ * |  |  ---------------------------ASStackLayoutSpec--------------------------  |  |
+ * |  |  |  -----------------ASStackLayoutSpec----------------                |  |  |
+ * |  |  |  |  --------------ASStackLayoutSpec-------------  |                |  |  |
+ * |  |  |  |  |  ASEditableTextNode  ASEditableTextNode  |  |                |  |  |
+ * |  |  |  |  --------------------------------------------  |                |  |  |
+ * |  |  |  |  --------------ASStackLayoutSpec-------------  |  ASButtonNode  |  |  |
+ * |  |  |  |  |  ASEditableTextNode  ASEditableTextNode  |  |                |  |  |
+ * |  |  |  |  --------------------------------------------  |                |  |  |
+ * |  |  |  --------------------------------------------------                |  |  |
+ * |  |  ----------------------------------------------------------------------  |  |
+ * |  ----------------------------------------------------------------------------  |
+ * |                                  ASButtonNode                                  |
+ * |                                    ASMapNode                                   |
+ * ----------------------------------------------------------------------------------
+ *
+ *  This diagram was created by setting a breakpoint on the returned `layoutSpec`
+ *  and calling "po [layoutSpec asciiArtString]" in the debugger.
+ */
 #define SPACING 5
 #define HEIGHT 30
-  CGSize nodeSize = CGSizeMake(constrainedSize.max.width * 0.3, HEIGHT);
-
-  _latEditableNode.style.preferredSize = nodeSize;
-  _lonEditableNode.style.preferredSize = nodeSize;
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+//  _latEditableNode.style.minWidth      = ASDimensionMake(@"30%");
+//  _lonEditableNode.style.minWidth      = ASDimensionMake(@"30%");
+//  _deltaLatEditableNode.style.minWidth = ASDimensionMake(@"30%");
+//  _deltaLonEditableNode.style.minWidth = ASDimensionMake(@"30%");
+
+  _latEditableNode.style.flexGrow      = YES;
+  _lonEditableNode.style.flexGrow      = YES;
+  _deltaLatEditableNode.style.flexGrow = YES;
+  _deltaLonEditableNode.style.flexGrow = YES;
   
-  _deltaLatEditableNode.style.preferredSize = nodeSize;
-  _deltaLonEditableNode.style.preferredSize = nodeSize;
+  _updateRegionButton.style.flexGrow = YES;
+  
+  _liveMapToggleButton.style.flexGrow = YES;
+  _liveMapToggleButton.style.maxHeight = ASDimensionMake(HEIGHT);
```

Comment:
I'm looking to make the height constrained for the "Live Map is OFF" button:

![image](https://cloud.githubusercontent.com/assets/3419380/19377378/bf39228a-919a-11e6-8007-5079f32492d6.png)